fdbserver/workloads/MetaclusterManagementWorkload.actor.cpp

```diff
@@ -750,19 +792,30 @@ struct MetaclusterManagementWorkload : TestWorkload {
 			ASSERT(assignedCluster != self->dataDbs.end());
 			ASSERT(assignedCluster->second->tenants.try_emplace(tenant, tenantData).second);
 
+			Optional<ClusterName> clusterAssignedToTenantGroup;
 			if (tenantGroup.present()) {
 				auto tenantGroupData =
 				    self->tenantGroups
 				        .try_emplace(tenantGroup.get(), makeReference<TenantGroupData>(entry.assignedCluster))
 				        .first;
 				ASSERT(tenantGroupData->second->cluster == entry.assignedCluster);
+				clusterAssignedToTenantGroup = tenantGroupData->second->cluster;
 				tenantGroupData->second->tenants.insert(tenant);
 				assignedCluster->second->tenantGroups[tenantGroup.get()] = tenantGroupData->second;
 			} else {
 				self->ungroupedTenants.insert(tenant);
 				assignedCluster->second->ungroupedTenants.insert(tenant);
 			}
 
+			// In two cases, we bypass the search on capacity index:
+			// i. assignClusterAutomatically is false. This indicates that the user explicitly specifies which cluster
+			//    to assign to a tenant
+			// ii. the tenant belongs to a tenant group to which a cluster has already been assigned.
+			// Except for these two cases, any chosen cluster must have autoTenantAssignment enabled.
+			ASSERT(!assignClusterAutomatically ||
+			       (clusterAssignedToTenantGroup.present() && !clusterAssignedToTenantGroup.get().empty()) ||
```

Comment:
Do we need the condition that the cluster name is not empty, or is it sufficient to check that the optional is non-empty?

Reply:
We actually need just one. Will remove the second condition.